conformance/lang/expressions/range-expr/range_expr_exclusive.balt

```diff
@@ -0,0 +1,5847 @@
+        optional-type, range-expr, record-type
+
+function init() {
+    int startIndex = 5;
+    int endIndex = 3;
+    int count = 0;
+    int[] arr = [];
+
+    object {
+        *object:Iterable;
+        public isolated function iterator() returns object {
+            public isolated function next() returns (record {|int value;|}?);
+        };
+    } range = startIndex ..< endIndex;
+
+    foreach int i in range {
+          arr.push(i);
+          count = count + 1;
+    }
+    io:println(count); // @output 0
+    io:println(arr); // @output []
+}
+
+Test-Case: output
+Description: Test range expression (exclusive) with iterable object when the types of the expressions are inferred.
+Labels: additive-expr, array-type, array:push, DecimalNumber, int, iterable-object, list-constructor-expr, object-type,
+        optional-type, range-expr, record-type, var
+
+function init() {
+    var startIndex = 1;
+    var endIndex = 10;
+    int count = 0;
+    int[] arr = [];
+
+    object {
+        *object:Iterable;
+        public isolated function iterator() returns object {
+            public isolated function next() returns (record {|int value;|}?);
+        };
+    } range = startIndex ..< endIndex;
+
+    foreach int i in range {
+          arr.push(i);
+          count = count + 1;
+    }
+    io:println(count); // @output 9
+    io:println(arr); // @output [1,2,3,4,5,6,7,8,9]
+}
+
+Test-Case: output
+Description: Test range expression (exclusive) when the types of the expressions and type of the range expression are inferred.
+Labels: additive-expr, array-type, array:push, DecimalNumber, int, iterable-object, list-constructor-expr, range-expr,
+        var
+
+function init() {
+    var startIndex = 1;
+    var endIndex = 10;
+    int count = 0;
+    int[] arr = [];
+
+    var range = startIndex ..< endIndex;
+
+    foreach int i in range {
+          arr.push(i);
+          count = count + 1;
+    }
+    io:println(count); // @output 9
+    io:println(arr); // @output [1,2,3,4,5,6,7,8,9]
+}
+
+Test-Case: output
+Description: Test range expression (exclusive) at module level.
+Labels: additive-expr, array-type, array:push, DecimalNumber, int, iterable-object, list-constructor-expr, object-type,
+        optional-type, range-expr, record-type, var
+
+int startIndex = 1;
+int endIndex = 10;
+
+object {
+    *object:Iterable;
+    public isolated function iterator() returns object {
+        public isolated function next() returns (record {|int value;|}?);
+    };
+} range = startIndex ..< endIndex;
+
+function init() {
+    int count = 0;
+    int[] arr = [];
+
+    foreach int i in range {
+          arr.push(i);
+          count = count + 1;
+    }
+    io:println(count); // @output 9
+    io:println(arr); // @output [1,2,3,4,5,6,7,8,9]
+}
+
+Test-Case: output
+Description: Test range expression (exclusive) at module level.
```

Comment:
may be, (because we have the same description in the two test cases)
```suggestion
Description: Test range expression (exclusive) at module level with var.
```